src/transformers/models/whisper/feature_extraction_whisper.py

```diff
@@ -312,4 +317,9 @@ def __call__(
         if return_tensors is not None:
             padded_inputs = padded_inputs.convert_to_tensors(return_tensors)
 
+        if return_token_timestamps is not None:
+            padded_inputs["num_frames"] = torch.tensor(
```

Comment:
This should use `.convert_to_tensor` to respect the tensor type requested by the user, not forcing it to a torch tensor